server/data/prisonApiClient.test.ts

```diff
@@ -1,247 +1,109 @@
-      expect(hmppsAuthClient.getSystemClientToken).toBeCalled()
-    })
-
-    it('Prisoner detail - error no admin token', async () => {
-      hmppsAuthClient.getSystemClientToken.mockResolvedValue('')
-      fakeApi.get('/api/offenders/AA1234A', '').reply(401)
-      try {
-        await prisonerService.getPrisonerDetail('XTEST1', 'AA1234A')
-      } catch (e) {
-        expect(e.message).toContain('Unauthorized')
-      }
-      expect(nock.isDone()).toBe(true)
-      expect(hmppsAuthClient.getSystemClientToken).toBeCalled()
-    })
-
-    it('Prisoner detail - not found', async () => {
-      hmppsAuthClient.getSystemClientToken.mockResolvedValue('a token')
-      fakeApi.get('/api/offenders/AA1234A', '').reply(404)
-      try {
-        await prisonerService.getPrisonerDetail('XTEST1', 'AA1234A')
-      } catch (e) {
-        expect(e.message).toContain('Not Found')
-      }
-      expect(nock.isDone()).toBe(true)
-      expect(hmppsAuthClient.getSystemClientToken).toBeCalled()
-    })
+  it('Get prisoner image', async () => {
+    stream.mockResolvedValue(Readable.from('image'))
+
+    const result = await prisonApiClient.getPrisonerImage('ABC1234', 'joebloggs')
+
+    expect(stream).toHaveBeenCalledWith({ path: '/api/bookings/offenderNo/ABC1234/image/data' }, 'joebloggs')
+    expect(result.read()).toEqual('image')
   })
 
-  describe('Prison information - admin token', () => {
-    const stubbedPrisonData = {
-      addressType: 'BUS',
-      agencyId: 'MDI',
-      agencyType: 'INST',
-      description: 'Moorland (HMP)',
-      formattedDescription: 'Moorland (HMP)',
-    } as PrisonInformation
-
-    it('Get prison information', async () => {
-      hmppsAuthClient.getSystemClientToken.mockResolvedValue('a token')
-      fakeApi.get('/api/agencies/prison/MDI').reply(200, stubbedPrisonData)
-      const result = await prisonerService.getPrisonInformation('XTEST1', 'MDI')
-      expect(result.agencyId).toEqual('MDI')
-      expect(nock.isDone()).toBe(true)
-      expect(hmppsAuthClient.getSystemClientToken).toBeCalled()
-    })
+  it('Get prisoner image as JPEG', async () => {
+    get.mockResolvedValue(Buffer.from('image'))
+
+    const result = await prisonApiClient.getPrisonerImageData('ABC1234', 'joebloggs')
+
+    expect(get).toHaveBeenCalledWith(
+      { path: '/api/bookings/offenderNo/ABC1234/image/data', responseType: 'image/jpeg' },
+      'joebloggs'
+    )
+    expect(result.toString()).toEqual('image')
   })
 
-  describe('Prisoner detail - access with user token', () => {
-    const stubbedPrisonUserData = {
-      accountStatus: 'ACTIVE',
-      active: true,
-      activeCaseLoadId: 'MDI',
-      expiredFlag: false,
-      firstName: 'Robert',
-      lastName: 'Charles',
-      lockedFlag: false,
-      staffId: 123,
-      thumbnailId: 345,
-      username: 'RCHARLES',
-    } as PrisonApiUserDetail
-
-    it('Get /api/users/me', async () => {
-      fakeApi.get('/api/users/me').reply(200, stubbedPrisonUserData)
```

Comment:
This one should pass through the logged-in users own token. Can't see the equivalent (yet)?

Reply:
Should be addressed now!